listenbrainz/webserver/static/js/src/types.d.ts

```diff
@@ -2,8 +2,43 @@ declare module "spotify-web-playback-sdk";
 declare module "react-bs-notifier";
 declare module "time-ago";
 
-// TODO: make this type specific
-declare type Listen = any;
+// TODO: Remove "| null" when backend stops sending fields with null
+declare type Listen = {
+  listened_at: number;
+  listened_at_iso?: string | null;
+  playing_now?: boolean | null;
+  user_name?: string | null;
+  track_metadata: {
+    artist_name: string;
+    release_name?: string | null;
+    track_name: string;
+    additional_info?: {
+      artist_mbids?: Array<string> | null;
+      artist_msid?: string | null;
+      discnumber?: number | null;
+      duration_ms?: number | null;
+      isrc?: string | null;
+      listening_from?: string | null;
+      recording_mbid?: string | null;
+      recording_msid?: string | null;
+      release_artist_name?: string | null;
+      release_artist_names?: Array<string> | null;
+      release_group_mbid?: string | null;
+      release_mbid?: string | null;
+      release_msid?: string | null;
+      spotify_album_artist_ids?: Array<string> | null;
+      spotify_album_id?: string | null;
+      spotify_artist_ids?: Array<string> | null;
+      spotify_id?: string | null;
+      tags?: Array<string> | null;
+      track_mbid?: string | null;
+      tracknumber?: number | null;
+      work_mbids?: Array<string> | null;
+      [key: string]: any;
```

Comment:
if you make additional info a seperate interface, you might not need this line.

Reply:
I did not understand how making `additional_info` into a separate interface will help solve the issue. Can you please explain how it can be done.